File: backend/app/api/sys_dict.py
from flask import request, jsonify

from app.api import api_bp
from app.services import sys_dict_service
from app.utils.decorators import token_required, admin_required


@api_bp.route('/sys_dicts', methods=['POST'])
@token_required
@admin_required
def create_sys_dict(current_user):
    data = request.get_json()
    sys_dict = sys_dict_service.create_sys_dict(data)
    return jsonify(sys_dict.to_dict()), 201


@api_bp.route('/sys_dicts/<int:id>', methods=['GET'])
@token_required
@admin_required
def get_sys_dict(current_user, id):
    sys_dict_data = sys_dict_service.get_sys_dict(id)
    return jsonify(sys_dict_data)


@api_bp.route('/sys_dicts/<int:id>', methods=['PUT'])
@token_required
@admin_required
def update_sys_dict(current_user, id):
    data = request.get_json()
    sys_dict = sys_dict_service.update_sys_dict(id, data)
    return jsonify(sys_dict.to_dict())


@api_bp.route('/sys_dicts/<int:id>', methods=['DELETE'])
@token_required
@admin_required
def delete_sys_dict(current_user, id):
    sys_dict_service.delete_sys_dict(id)
    return '', 204


@api_bp.route('/sys_dicts', methods=['GET'])
@token_required
@admin_required
def get_sys_dicts(current_user):
    return jsonify(sys_dict_service.get_sys_dict_all())

File: frontend/src/api/sys_dict.ts
import request from './request'
import type { SysDictInput } from '../schemas/sys_dict'

/** 后端契约：系统字典项；id 创建前可空 */
export interface SysDict {
  id?: number
  key: string
  value: string
  des?: string
  enable: boolean
}

/** 列出全部系统字典 */
export function getSysDicts() {
  return request.get<SysDict[]>('/sys_dicts')
}

/** 获取单条字典 */
export function getSysDict(id: number) {
  return request.get<SysDict>(`/sys_dicts/${id}`)
}

/** 创建字典项 */
export function createSysDict(data: SysDictInput) {
  return request.post<SysDict>('/sys_dicts', data)
}

/** 更新字典项 */
export function updateSysDict(id: number, data: SysDictInput) {
  return request.put<SysDict>(`/sys_dicts/${id}`, data)
}

/** 删除字典项 */
export function deleteSysDict(id: number) {
  return request.delete<void>(`/sys_dicts/${id}`)
}
